crates/ty_python_semantic/src/semantic_index/builder.rs

```diff
@@ -1562,6 +1567,15 @@ impl<'ast> Visitor<'ast> for SemanticIndexBuilder<'_, 'ast> {
                         (&alias.name.id, false)
                     };
 
+                    // If there's no alias or a redundant alias, record this as a potential import of a submodule
+                    if alias.asname.is_none() || is_reexported {
```

Comment:
Would it be sufficient to only collect the import statements at the module level (skip if we're inside a function or class)? In which case, I even wonder if this has to be inside `semantic_index` or if it could just be its own salsa query that traverses the top-level statements of a module without traversing into inner blocks.

Reply:
I was going off of the fact that we apply no such logic when handling `imported_modules`. I agree anything but top-level feels a biiit dubious (and I believe pyright agrees with that).